include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+
+struct otTcpListener;
+typedef struct otTcpListener otTcpListener;
+
+#define OT_TCP_DEFER  ((otTcpEndpoint *) NULL)
+#define OT_TCP_REFUSE ((otTcpEndpoint *) -1)
+
+/**
+ * This callback indicates that an incoming connection that matches this
+ * TCP listener has arrived. The application should provide (via the callback's
+ * return value) a pointer to the @p otTcpEndpoint into which to accept the
+ * incoming connection. The returned @p otTcpEndpoint must already be
+ * initialized (i.e., @p otTcpEndpointInitialize must have been called on it).
+ *
+ * The application may instead choose to return one of two special values.
+ * If the application returns OT_TCP_DEFER, then OpenThread silently ignores
+ * the connection establishment request; the callback will be called again the
+ * next time that the connection establishment request is submitted. This is
+ * valuable if resources are not available to accept the connection, but they
+ * may be available at the time of a future attempt. If the application returns
+ * OT_TCP_REFUSE, then OpenThread sends a "connection refused" message to the
+ * host that attempted to establish a connection.
+ */
+typedef otTcpEndpoint *(*otTcpAcceptReady)(void *aContext, otTcpListener *aListener, const otSockAddr *aPeer);
+
+/**
+ * This callback indicates that the TCP connection is now ready for two-way
+ * communication. In the case of TCP Fast Open, this may be before the TCP
+ * connection handshake has actually completed. The application is provided
+ * with the context pointers both for the TCP listener that accepted the
+ * connection and the TCP endpoint into which it was accepted. The provided
+ * context is the one associated with the TCP listener.
+ */
+typedef void (*otTcpAcceptDone)(void *aContext, otTcpListener *aListener, otTcpEndpoint *aEndpoint, const otSockAddr *aPeer);
+
+/**
+ * This structure represents a TCP listener.
+ *
+ * A TCP listener is used to listen for and accept incoming TCP conections.
+ */
+typedef struct otTcpListener
+{
+    struct otTcpListener *mNext;            ///< A pointer to the next TCP listener (internal use only)
+    void *mContext;                         ///< A pointer to application-specific context
+    otTcpAcceptReady mAcceptReadyCallback;  ///< "Accept ready" callback function
+    otTcpAcceptDone mAcceptDoneCallback;    ///< "Accept done" callback function
+
+    /* Other implementation-defined fields go here. */
+} otTcpListener;
+
+/**
+ * This structure contains arguments to the @p otTcpListenerInitialize
+ * function.
+ */
+typedef struct otTcpListenerInitializeArgs
+{
+    otTcpAcceptReady mAcceptReadyCallback;  ///< "Accept ready" callback function
+    otTcpAcceptDone mAcceptDoneCallback;    ///< "Accept done" callback function
+    void *mContext;                         ///< Pointer to application-specific context
+} otTcpListenerInitializeArgs;
+
+/**
+* Initializes a TCP listener. Calling this function causes OpenThread to keep
```

Comment:
nit: have brief in separate block above

Reply:
Done. I'm not 100% what you meant by "in separate block", can you please check that what I've done is acceptable?